snap/snapcraft.yaml

```diff
@@ -0,0 +1,77 @@
+name: ghostty
+base: core24
+version: git
+summary: A terminal emulator
+description: |
+  Ghostty is a fast, feature-rich, and cross-platform terminal emulator that
+  uses platform-native UI and GPU acceleration.
+grade: stable
+confinement: classic
+contact: https://github.com/ghostty-org/ghostty/discussions
+issues: https://github.com/ghostty-org/ghostty/issues
+website: https://ghostty.org
+license: MIT
+icon: images/icons/icon_512.png
+
+platforms:
+  amd64:
+  arm64:
+
+apps:
+  ghostty:
+    command: bin/ghostty
+    command-chain: [ bin/launcher ]
+    desktop: share/applications/com.mitchellh.ghostty.desktop 
+    environment:
+      PATH: /snap/ghostty/current/bin:/snap/ghostty/current/usr/bin:$PATH
+      LC_ALL: C.UTF-8
+
+parts:
+  launcher:
+    plugin: dump
+    source: snap/local/
+    organize:
+      launcher: bin/
+
+  zig:
+    plugin: nil
+    build-packages:
+      - curl
+    override-pull: |
+      set -ex
+      case "$CRAFT_TARGET_ARCH" in
+        amd64)  arch=x86_64    ;;
+        arm64)  arch=aarch64  ;;
+        *)      arch=""     ;;
```

Comment:
Presumably it would be good to exit non-zero in this case with a good error message before curl tries to download something that doesn't exist.